src/esp/physics/bullet/BulletRigidObject.cpp

```diff
@@ -212,7 +210,7 @@ bool BulletRigidObject::initializeObject(
   }
 
   //! Bullet rigid body setup
-  bObjectMotionState_ = new Magnum::BulletIntegration::MotionState(*this);
+  bObjectMotionState_ = new Magnum::BulletIntegration::MotionState(node());
```

Comment:
What about deriving `BulletRigidObject` from `MotionState`? Since those two are tightly coupled anyway.

As it is now, creating a `BulletRigidObject` and then destructing it again would leave the (dangling) MotionState feature still attached to the original object. I think both should be removed, and the easiest way to achieve that is to derive from it.